pkg/netpol/connlist/exposure_analysis.go

```diff
@@ -0,0 +1,151 @@
+}
+
+func (ep *exposedPeer) EgressExposure() []XgressExposureData {
+	return xgressExposureListToXgressExposureDataList(ep.egressExp)
+}
+
+// ----------------------------------------------------
+
+// gets an exposure map and builds ExposedPeer list while refining exposure connections
+// for each peer refines its exposure data as followed:
+// * if a peer is not protected by netpols on any direction (ingress/egress) :
+// the xgress lists are already empty so will have an empty xgressExp list (output will display not protected)
+// * if a peer is protected and exposed to entire cluster on all conns:
+// the list will include only this (eliminates other conns with specific namespaces)
+// * if a peer is exposed to entire cluster on ingress/egress on a specific connection:
+// eliminate exposure-data to specific potential namespaces on same direction and same connection
+func buildExposedPeerListFromExposureMap(exposuresMap exposureMap) []ExposedPeer {
+	res := make([]ExposedPeer, 0)
+	for p, expData := range exposuresMap {
+		ingExp := make([]*xgressExposure, 0)
+		egExp := make([]*xgressExposure, 0)
+		if expData.isIngressProtected {
+			ingExp = append(ingExp, loopAndRefineXgressData(expData.ingressExposure)...)
+		}
+		if expData.isEgressProtected {
+			egExp = append(egExp, loopAndRefineXgressData(expData.egressExposure)...)
+		}
+		// final peer's exposure data
+		expInfo := &exposedPeer{
+			peer:               p,
+			isIngressProtected: expData.isIngressProtected,
+			isEgressProtected:  expData.isEgressProtected,
+			ingressExp:         ingExp,
+			egressExp:          egExp,
+		}
+		res = append(res, expInfo)
+	}
+	return res
+}
+
+func loopAndRefineXgressData(xgressData []*xgressExposure) []*xgressExposure {
+	res := make([]*xgressExposure, 0)
+	// helping var
+	var entireClusterConn common.AllowedConnectivity
+	for _, singleConn := range xgressData {
+		//  exposed to entire cluster on all conns - result will include this one general exposureConn
+		if singleConn.exposedToEntireCluster && singleConn.potentialConn.AllProtocolsAndPorts() {
+			res = nil // remove previous conns if exist
+			res = append(res, singleConn)
+			break
+		}
+		if singleConn.exposedToEntireCluster {
+			entireClusterConn = singleConn.potentialConn
+			// refine result - exclude data to/from specific ns with same conn value
+			res = nil
+			res = refineConnsWithSameValueFromRes(res, entireClusterConn)
+		}
+		// exposed to specific namespace with same connection exposed to any-namespace , skip
+		if !singleConn.exposedToEntireCluster && singleConn.potentialConn == entireClusterConn {
+			continue
+		}
+		res = append(res, singleConn)
+	}
+	return res
+}
+
+// returns the list without items with provided conn value
+func refineConnsWithSameValueFromRes(expList []*xgressExposure, conn common.AllowedConnectivity) []*xgressExposure {
+	res := make([]*xgressExposure, 0)
+	for _, singleConn := range expList {
+		if singleConn.potentialConn != conn {
```

Comment:
- in golang , comparing structs with `==` , compares each field separately.
I tried it, since worked with the (3) example above; kept it 

will change and add an containment func to the `allowedConnectivity` interface :( if the rep. ns's connection equal or contained in the `any-namespace` conns : so it will be refined)

Reply:
why not having internally the `common.ConnectionSet` type that already implements containment function?

Reply:
i am now aware of it;  and saw that it implements the `common.Connection` interface  which is also the type of `allowedConnections` .
so Can I also use the `common.Connection` interface as the type of the `PotentialConnectivity()` field in `XgressExposureData` interface 

(then will revert the `AllowedConnectivity` interface movement from `diff` pkg)

Reply:
why do we have both `AllowedConnectivity` and `Connection` interfaces in package `common`?

Reply:
I think the goal for defining `AllowedConnectivity ` in the diff package was that this type was returned through the API, so wanted to avoid returning object of type from internal pkg.

Reply:
maybe have another "connection" pkg (not internal) that exposes the interfaces for `AllowedConnectivity` , to be used for both `connlist` and `diff` packages? 
Internally should use the `common.ConnectionSet` for the operations between objects.

Reply:
I see , can we eliminate `AllowedConnectivity` and put the `Connection` interface in a new package ? (seems to me a code dup. to have both?)
and so internally we use `common.ConnectionSet` as the struct implementing it in both packages?

Reply:
yes, I think so

Reply:
- why not using internally  (in our structs) the explicit `ConnectionSet` struct, and use `Connection` interface only in the exposed interfaces? 
- please open a separate issue about diff allowedConnectivity struct and connlist connection struct.

Reply:
* done [commit](https://github.com/np-guard/netpol-analyzer/pull/296/commits/9e154f075958551c99f4a006a3c20b9b0d5914ea)
* added new issue #297